app/code/Magento/CatalogSearch/Model/Search/FilterMapper/StockJoinProviderInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\CatalogSearch\Model\Search\FilterMapper;
+
+use Magento\Framework\DB\Select;
+
+/**
+ * SPI Interface to change the stock join
```

Comment:
description of the interface looks awkward.

It's not clear the main responsibility of this entity and why we need it